validator/index.ts

```diff
@@ -0,0 +1,48 @@
+import blgr from 'blgr';
+import ethers from "ethers";
+import to from 'await-to-js';
+import {ValidatorCtx} from "./types";
+import {unlockWallet} from "./utils/wallet";
+const logger = blgr.logger('debug');
+
+export class Validator {
+  private ctx: ValidatorCtx;
+  private publicWallet: ethers.Wallet;
+  private withdrawalWallet: ethers.Wallet;
+  constructor(ctx: ValidatorCtx) {
+    this.ctx= ctx;
+  }
+
+  setupRPC(): void {
+    logger.info("Setting up RPC connection...");
+    // TODO connect to a beacon chain
+    logger.info(`RPC connection successfully established ${this.ctx.rpcUrl}!`);
+  }
+
+  async validateKeystores() {
+    logger.info("Unlocking wallets...");
+
+    // Attempt to unlock public wallet
+    const [err1, publicWallet] = await to<ethers.Wallet>(unlockWallet(this.ctx.publicKeystore, this.ctx.publicKeystorePassword, "public wallet"));
+    if (err1) throw new Error("Public wallet could not be unlocked!");
+    this.publicWallet = publicWallet;
+
+    // Attempt to unlock withdrawal wallet
+    let [err2, withdrawalWallet] = await to<ethers.Wallet>(unlockWallet(this.ctx.withdrawalKeystore, this.ctx.withdrawalKeystorePassword, "withrawal wallet"));
```

Comment:
I think we may not want to do this. The spec mentions that the withdrawal wallet can stay in "cold storage".

Reply:
hmm good point. I'm thinking we can validate that the wallet is correct? Perhaps I destroy the object in mem?

Reply:
I'm thinking most people would probably have this key offline until they want to withdraw. We probably should only import/validate/use the wallet once a withdrawal is triggered.

eg:
```javascript
withdraw(keystore, password, etc... ) {
  // create withdrawal wallet
  ...
  // do withdraw
  ...
  // destroy withdrawal wallet
  ...
}
```

Reply:
Hmm ok. That works. I'm pretty sure we need to know the withdrawal address in advance let me double check

Reply:
Yeah deposit info Requirea withdrawal credentials... 

My concern is about safety ensuring they used the correct public address

Reply:
I think the DepositInput stores a hash of the public key of a 'withdrawal account'.
That sort of thing will definitely need to be checked, but I think its more appropriate when the user initially inputs their deposit on the eth1 chain.

At this point, the deposit on the eth1 chain has presumably already cleared, and if they didn't put the right withdrawal key, or lost it somewhere along the way, there's nothing we can do for them.

Reply:
Hmmmmmmmmmmmmmmm ok Im good with that assumption. I'll clean this up and redo the withrawal steps